### PR TITLE
Xw 310

### PR DESCRIPTION
**Pull request by justnpT**
test case 2
https://wikia-inc.atlassian.net/browse/XW-306

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    for (WidgetPageObject widget : widgets) {
+      Assertion.assertTrue(widget.isLoadedOnMercury(), MercuryMessages.INVISIBLE_MSG);
+    }
+  }
+
+  @Test(groups = "MercuryWidgetsCombinedTest_002")
+  @Execute(onWikia = "mercuryautomationtesting")
+  public void MercuryWidgetsCombinedTest_002_isLoadedOnFirstVisitFromDifferentArticle() {
+    String content = "";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java]: MAJOR Remove this unused "content" local variable. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    for (WidgetPageObject widget : widgets) {
+      Assertion.assertTrue(widget.isLoadedOnMercury(), MercuryMessages.INVISIBLE_MSG);
+    }
+  }
+
+  @Test(groups = "MercuryWidgetsCombinedTest_003")
+  @Execute(onWikia = "mercuryautomationtesting")
+  public void MercuryWidgetsCombinedTest_003_isLoadedOnSecondVisitFromDifferentArticle() {
+    String content = "";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java]: MAJOR Remove this unused "content" local variable. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 4 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/a23e7327aee5d5a880a26eb77974d86212e94743/src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/a23e7327aee5d5a880a26eb77974d86212e94743/src/test/java/com/wikia/webdriver/testcases/widgettests/WidgetsCombinedTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    for (WidgetPageObject widget : widgets) {
+      content += widget.getTag();
+    }
+
+    ArticleContent articleContent = new ArticleContent();
+    articleContent.clear(ARTICLE_NAME);
+    articleContent.push(content, ARTICLE_NAME);
+
+    ArticlePageObject article = new ArticlePageObject(driver);
+    article.open(ARTICLE_NAME);
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java]: article is used once, use `new ArticlePageObject(driver).openMercuryArticleByNameWithCbAndNoAds(ARTICLE_NAME)`

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  public void MercuryWidgetsCombinedTest_002_isLoadedOnFirstVisitFromDifferentArticle() {
+    String content = "";
+    for (WidgetPageObject widget : widgets) {
+      content += widget.getTag();
+    }
+
+    ArticlePageObject article = new ArticlePageObject(driver);
+    BasePageObject base = new BasePageObject(driver);
+    base.openMercuryArticleByNameWithCbAndNoAds(wikiURL, MercurySubpages.MAIN_PAGE);
+    article.open(ARTICLE_NAME);
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java]: those 4 lines can be replaced by 2
 `new ArticlePageObject(driver).openMercuryArticleByNameWithCbAndNoAds(MercurySubpages.MAIN_PAGE)`
 `new ArticlePageObject(driver).openMercuryArticleByNameWithCbAndNoAds(ARTICLE_NAME)`

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    String content = "";
+    for (WidgetPageObject widget : widgets) {
+      content += widget.getTag();
+    }
+
+    ArticlePageObject article = new ArticlePageObject(driver);
+    article.open(ARTICLE_NAME);
+    BasePageObject base = new BasePageObject(driver);
+    base.openMercuryArticleByNameWithCbAndNoAds(wikiURL, MAPS_ARTICLE_NAME);
+    article.open(ARTICLE_NAME);
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java]: those 5 lines can be replaced by 3
 `new ArticlePageObject(driver).openMercuryArticleByNameWithCbAndNoAds(ARTICLE_NAME)`
 `new ArticlePageObject(driver).openMercuryArticleByNameWithCbAndNoAds(MAPS_ARTICLE_NAME)`
 `new ArticlePageObject(driver).openMercuryArticleByNameWithCbAndNoAds(ARTICLE_NAME)`

File: src/test/java/com/wikia/webdriver/testcases/widgettests/WidgetsCombinedTests.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    for (WidgetPageObject widget : widgets) {
+      content += widget.getTag();
+    }
+
+    ArticleContent articleContent = new ArticleContent();
+    articleContent.clear(ARTICLE_NAME);
+    articleContent.push(content, ARTICLE_NAME);
+
+    ArticlePageObject article = new ArticlePageObject(driver);
+    article.open(ARTICLE_NAME);
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/widgettests/WidgetsCombinedTests.java]: article is used once, use `new ArticlePageObject(driver).openMercuryArticleByNameWithCbAndNoAds(ARTICLE_NAME)`

**Comment by tomasznapieralski** [comment]
Add file to test suite

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+    for (WidgetPageObject widget : widgets) {
+      Assertion.assertTrue(widget.isLoadedOnMercury(), MercuryMessages.INVISIBLE_MSG);
+    }
+  }
+
+  @Test(groups = "MercuryWidgetsCombinedTest_002")
+  @Execute(onWikia = "mercuryautomationtesting")
+  public void MercuryWidgetsCombinedTest_002_isLoadedOnFirstVisitFromDifferentArticle() {
+    String content = "";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java]: MAJOR Remove this unused "content" local variable. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+    for (WidgetPageObject widget : widgets) {
+      Assertion.assertTrue(widget.isLoadedOnMercury(), MercuryMessages.INVISIBLE_MSG);
+    }
+  }
+
+  @Test(groups = "MercuryWidgetsCombinedTest_003")
+  @Execute(onWikia = "mercuryautomationtesting")
+  public void MercuryWidgetsCombinedTest_003_isLoadedOnSecondVisitFromDifferentArticle() {
+    String content = "";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java]: MAJOR Remove this unused "content" local variable. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 4 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/c105fe3d57c995f1fd1bff4e5944d2e6f59b19a2/src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/c105fe3d57c995f1fd1bff4e5944d2e6f59b19a2/src/test/java/com/wikia/webdriver/testcases/widgettests/WidgetsCombinedTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/cd11dd1fa27712be3007c385a3cc9f52372c1678/src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/AllTagsWidgetTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/cd11dd1fa27712be3007c385a3cc9f52372c1678/src/test/java/com/wikia/webdriver/testcases/widgettests/AllTagsTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/9799352ce39bfc38bf5bb78ef63f5a48e3460136/src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/AllTagsWidgetTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/9799352ce39bfc38bf5bb78ef63f5a48e3460136/src/test/java/com/wikia/webdriver/testcases/widgettests/AllTagsTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/a5aaed87b60b875a16824bd82d93f46645a103b7/src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/AllTagsTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/a5aaed87b60b875a16824bd82d93f46645a103b7/src/test/java/com/wikia/webdriver/testcases/widgettests/AllTagsTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

**Comment by tomasznapieralski** [comment]
all tests are passing

**Comment by bognix** [comment]
LGTM

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/a5aaed87b60b875a16824bd82d93f46645a103b7/src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/AllTagsTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/a5aaed87b60b875a16824bd82d93f46645a103b7/src/test/java/com/wikia/webdriver/testcases/widgettests/AllTagsTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

**Pull request merged**